### DISCUSSION TITLE
Thematic attributes & codelists

### DISCUSSION THREAD

**Discussion by MayteTDGeograma** [gopeg/issues-barriers#42] [General]
It would be desirable to have tools to facilitate the extension of INSPIRE application schemas as well as the generation of the codelist for the new attributes (extended).

This will ease the adaptation of the schema (of an INSPIRE theme) to the user’s needs. 

The codelist, for the extended attributes field(s), would unify the possible records and, therefore, facilitate the data analysis.

Suppose an extended INSPIRE Building model, in which the population attribute has been added. A codelist with the origin of population data would be desirable to have correct traceability of this data.

**Comment by thorsten-reitz** [comment]
For methods and tutorials, there is a lot of informatiion here: https://inspire-extensions.wetransform.to/

Also, there are several tools which can help creating extended data models, such as Enterprise Architect of haleconnect (See https://help.wetransform.to/docs/setup-hc/create-a-schema/2018-01-28-schema-create-existing)

For extending codelists, a self-hosted re3gistry can be used. For more information, see https://github.com/ec-jrc/re3gistry/

**Comment by AndersOstmanNovogit** [comment]
This issue is described in section 4.2.7 in D3.4.

I have no additional solutions to add.
@MayteTDGeograma : Which solution did you apply?

**Reply by AndersOstmanNovogit** [reply]
Now reported in D4.3 section 4.2.3

**Comment by MayteTDGeograma** [comment]
For  PopImpact's buildings we have included new attributes in postGIS to collect population data.

We have used the manuals discussed by Thorsen to learn how to extend the models: https://inspire-extensions.wetransform.to
